From 3ce7c537258e26147011e7aaff42be5a37d1bdd2 Mon Sep 17 00:00:00 2001
From: ganeshi4u <ganeshvarmai4u@gmail.com>
Date: Fri, 6 Sep 2019 18:31:15 +0200
Subject: [PATCH 16/17] envsetup: Automatically set CCACHE_EXEC to the system's
 ccache

* check for ccache tool and set its path in CCACHE_EXEC if it hasn't been set already

Change-Id: Ife9d0a723cb90b89392ee288ebe6e7e9f2be7eef
---
 envsetup.sh | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/envsetup.sh b/envsetup.sh
index 089e17ff73..257d8b33b3 100644
--- a/envsetup.sh
+++ b/envsetup.sh
@@ -1755,4 +1755,15 @@ validate_current_shell
 source_vendorsetup
 addcompletions
 
+# check and set ccache path on envsetup
+if [ -z ${CCACHE_EXEC} ]; then
+    ccache_path=$(which ccache)
+    if [ ! -z "$ccache_path" ]; then
+        export CCACHE_EXEC="$ccache_path"
+        echo "ccache found and CCACHE_EXEC has been set to : $ccache_path"
+    else
+        echo "ccache not found/installed!"
+    fi
+fi
+
 export ANDROID_BUILD_TOP=$(gettop)
-- 
2.17.1

